src/cli/generate-idlparsed.js

```diff
@@ -0,0 +1,155 @@
+ *
+ * where `crawl report` is the path to the folder that contains the
+ * `index.json` file and all other crawl results produced by specs-crawler.js,
+ * and `save folder` is an optional folder (which must exist) where IDL
+ * name extracts are to be saved. In the absence of this parameter, the report
+ * is written to the console.
+ *
+ * When a folder is provided, the IDL name extracts are saved as a JSON
+ * structure in an `idlparsed` subfolder.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const webidlParser = require('../cli/parse-webidl');
+const {
+  expandCrawlResult,
+  requireFromWorkingDirectory
+} = require('../lib/util');
+
+
+/**
+ * Update the spec object in place with parsed IDL information.
+ * 
+ * @function
+ * @public
+ * @param {Object} spec The spec object to update. The function looks for the
+ *   raw IDL in the `idl` property.
+ * @return {Object} The updated spec with an `idl` property that contains the
+ *   parsed version of the IDL, and the raw IDL moved under the `idl.idl`
+ *   sub-property. Note the spec object is updated in place.
+ */
+async function generateIdlParsed(spec) {
+  if (!spec?.idl) {
+    return spec;
+  }
+  const rawIdl = spec.idl.idl ?? spec.idl;
+  try {
+    const parsedIdl = await webidlParser.parse(rawIdl);
+    parsedIdl.hasObsoleteIdl = webidlParser.hasObsoleteIdl(rawIdl);
+    parsedIdl.idl = rawIdl;
+    spec.idl = parsedIdl;
+  }
+  catch (err) {
+    // IDL content is invalid and cannot be parsed.
+    // Let's return the error, along with the raw IDL
+    // content so that it may be saved to a file.
+    err.idl = rawIdl;
+    spec.idl = err;
+  }
+  return spec;
+}
+
+
+async function generateIdlParsedFromPath(crawlPath) {
+  const crawlIndex = requireFromWorkingDirectory(path.resolve(crawlPath, 'index.json'));
+  const crawlResults = await expandCrawlResult(crawlIndex, crawlPath, ['idl']);
+  await Promise.all(crawlResults.results.map(generateIdlParsed));
+  return crawlResults;
+}
+
+
+async function createFolderIfNeeded(name) {
+  try {
+    await fs.promises.mkdir(name);
+  }
+  catch (err) {
+    if (err.code !== 'EEXIST') {
+      throw err;
+    }
+  }
+}
+
+
+/**
+ * Generate the `idlparsed` export for the spec.
+ * 
+ * Note that the raw IDL (under `spec.idl.idl`) gets deleted in the process.
+ *
+ * @function
+ * @public
+ * @param {Object} spec Spec object with the parsed IDL
+ * @param {String} folder Path to root folder where `idlparsed` folder needs to
+ *   appear.
+ * @return {String} The relative path from the root folder to the generated file
+ */
+async function saveIdlParsed(spec, folder) {
+  function specInfo(spec) {
+    return {
+      spec: {
+        title: spec.title,
+        url: spec.crawled
+      }
+    };
+  }
+
+  const subfolder = path.join(folder, 'idlparsed');
+  await createFolderIfNeeded(subfolder);
+
+  if (!spec?.idl?.idl) {
+    return;
+  }
+
+  delete spec.idl.idl;
```

Comment:
wouldn't it be cleaner to delete it only in the generated object? it feels a bit foot-gunny to modify the original object in a function called "save"

Reply:
Yes, that requires making at least a shallow copy of the object, which the code currently does not do.

The cleanest option would be to stop creating a hybrid structure under `idl` that holds both the result of parsing the raw IDL *and* the raw IDL (under an `idl` property), especially since that's clumsy and we need to separate them in the rest of the code in any case. The parsed IDL should rather be stored in a separate `idlparsed` property. It is a breaking change for code that use Reffy to do further processing on crawl results (e.g. Strudy), but it's easy to update code that relies on this, so this update seems a good opportunity to shave that yak...